pipelines/build/common/riscv_version_out.groovy

```diff
@@ -0,0 +1,116 @@
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+/**
+ * This file is a jenkins job for extracting a version string from a cross-compiled, riscv binary.
+ * See https://github.com/AdoptOpenJDK/openjdk-build/issues/1773 for the inspiration for this.
+ *
+ * This file is referenced by the upstream job at pipelines/build/common/openjdk_build_pipeline.groovy.
+ * This job is run at build-scripts/job/utils/job/riscv-version-out/.
+ *
+ * The job looks like:
+ *  1. Switch into a riscv node
+ *  2. Retrieve the artifacts built in the upstream job
+ *  3. Run java -version and export to file
+ *  4. Expose to the upstream job by archiving file
+ */
+
+// TODO: ADD THE ACTIVE NODE TIMEOUT LOGIC HERE OR GET IT MERGED INTO JOB HELPER (https://github.com/AdoptOpenJDK/openjdk-build/issues/2235)
+
+node ("riscv&&ci.role.test") {
```

Comment:
The build machines used for cross compiling shouldn't have the `riscv` tag on them (but as discused elsewhere they seem to at the moment) but we can roll with this for now and update it later

Reply:
That's probably worth opening an issue for, as it's worrying they have been labelled as such when they actually aren't

Reply:
@M-Davies I can see machines with `riscv64` but not `riscv` - did you hit some problem using `riscv`? I've removed `riscv64` from one of the static machines (which is offline anyway) but I'm surprised that it caused a problem with the current machine sets and labels

Reply:
I don't think I tried `riscv64` as I didn't realise it existed. The issue I had with `riscv` on it's own was that it always `node` into an azure build agent instead of an actual riscv machine. Adding `ci.role.test` ensured it picked up one of George's machines